build.gradle.kts

```diff
@@ -99,9 +98,8 @@ dependencies {
     testImplementation(gradleApi())
 }
 
-java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+kotlin {
+    jvmToolchain(8)
```

Comment:
While this is nicer, it has a big side effect: many machines need to download a separate JDK just to compile 8. Even JDK 21 supports targeting 8!

Reply:
I guess stutter downloads them anyway (when running tests), so it's fine.

Reply:
> I guess stutter downloads them anyway (when running tests), so it's fine.

However, not everyone runs the tests with stutter for JDK 8. In #298 I propose a concept to build (and run tests) with the current version by default. Only the regression tests require other JDK, but it is handled by CI (or can be triggered by a developer, if needed). In that context the original assignment seems more flexible to a regular developer build.

Reply:
True, we could also argue that since the foojay resolver is pretty common in Gradle builds, the availability of all relevant JDKs is likely on any dev machine doing Gradle-based dev. Even if not, it's an almost negligible one-time cost. If this was actively maintained by hundreds or had cost-sensitive uncached CI, my view would be different.

---

That said, it's probably better to use the default JVM with target 8 as before, so that the target version is not *implicit* from the toolchain. It also reduces any potential contributor's setup complexity.

So the question is what we optimise for? We seem to have a consensus for DX with @szpak. @3flex I'm curious what's are your thoughts on the above points.

Reply:
> so that the target version is not implicit from the toolchain

That's a feature, not a bug! And I don't think "setup complexity" is an issue here, the whole point of toolchains is that a user doesn't have to muck around with JDKs when toolchains are in use. It's a one-off download if JDK 8 isn't already on the machine. After that things "just work".

Toolchains are prominently featured in the Kotlin docs for configuring a Gradle project so I don't think this config will surprise anyone.

It's not hard to remove use of the toolchain here though if it's not wanted.

Reply:
@3flex What is a "problem" with using the current approach with building the project (by default) with the JDK that is used to start Gradle?

Support for building with JDK 8 is [deprecated](https://github.com/gradle-nexus/publish-plugin/pull/303), so I don't know, if forcing users to have (or automatically download JDK 8) matches the aforementioned.

Reply:
It's not a problem. It just makes the config simpler and in line with best practice.

I'm going to remove the use of toolchains then to get this over the line.